XMLSec1 testing

gasbytes opened the pull request:
- add lenient DER parser for ECDSA signatures when we fail with -171, so
  that we extract the r and s parameters, when the der is malformed when the integer
  values are raw bytes without proper sign-bit handling (those ders are
  used when signing/veriffing in the xmlsec1's testsuite);
- handle 0-byte cipher finalization in aes-cbc, we return -173 when data
  size = 0 when decrypting, this gets called by xmlsec1 to flush the
  buffer after decryption is done (not needed for wolfssl, but we return 0 in that case);
- Added v1.21.4 to the wget workflow (running on native ubuntu 24.04);
- Added v2.6.7 to the openldap workflow (running on native ubuntu 24.04);
- Added wireshark workflow (running on native ubuntu 24.04);
- Added xmlsec workflow (running on native ubuntu 24.04);

gasbytes committed:
- add lenient DER parser for ECDSA signatures when we fail with -171, so
  that we extract the r and s parameters, when the der is malformed when the integer
  values are raw bytes without proper sign-bit handling (those ders are
  used when signing/veriffing in the xmlsec1's testsuite);
- handle 0-byte cipher finalization in aes-cbc, we return -173 when data
  size = 0 when decrypting, this gets called by xmlsec1 to flush the
  buffer after decryption is done (not needed for wolfssl, but we return 0 in that case);

gasbytes committed:
- Added v1.21.4 to the wget workflow
- Added v2.6.7 to the openldap workflow

gasbytes committed:
Run wget-v1.21.4 and openldap-v2.6.7 on native ubuntu:24.04 instead of
debian:bookworm

SparkiDev commented on wolfssl-gnutls-wrapper/src/cipher.c:
```diff
@@ -993,6 +993,13 @@ int wolfssl_cipher_decrypt(void *_ctx, const void *src, size_t src_size,
         }
 
+	/* Handle 0-byte finalization call, common in cipher APIs for flushing/padding */
+	if (src_size == 0) {
```
Please don't use TABs for indenting.

SparkiDev commented on wolfssl-gnutls-wrapper/src/pk.c:
```diff
@@ -1777,6 +1777,110 @@ static int wolfssl_pk_verify_rsa_pss(const gnutls_datum_t *vdata,
+    if (len & 0x80) {
+        /* Long form length */
+        word32 num_bytes = len & 0x7F;
+        if (num_bytes > 4 || idx + num_bytes > sig_len) return -1;
```
When len is 0x80 then this is indefinite length encoded meaning go look for EOC.
Don't want to support that here.

SparkiDev commented on wolfssl-gnutls-wrapper/src/pk.c:
```diff
@@ -1777,6 +1777,110 @@ static int wolfssl_pk_verify_rsa_pss(const gnutls_datum_t *vdata,
+    /* Parse r length */
+    if (idx >= sig_len) return -1;
+    len = sig_data[idx++];
+    if (len & 0x80) {
```
Extract into a function as this code is required in multiple places.

SparkiDev commented on wolfssl-gnutls-wrapper/src/pk.c:
```diff
@@ -1777,6 +1777,110 @@ static int wolfssl_pk_verify_rsa_pss(const gnutls_datum_t *vdata,
+    }
+
+    /* Skip leading zero byte if present (sign byte) */
+    if (len > 0 && idx < sig_len && sig_data[idx] == 0x00) {
```
len must be greater than 0.

gasbytes committed:
- extracted parsing of DER length into a separate function;
- removed accidental Tab in the wolfssl-gnutls-wrapper/src/cipher.c
  file;

gasbytes committed:
- Added proper configurations for versions in wget and openldap that are
being tested  on ubuntu 24.04.
- Wireshark workflow
- Removing master version testing from the libjcat workflow.
Uses post quantum by default, not currently supported by gnutls-wolfssl.

gasbytes committed:
xmlsec workflow

gasbytes committed:
- return statesments on new lines;
- return -1 on num_bytes = 0 (unsupported);
